src/Cli/func/Helpers/PythonHelpers.cs

```diff
@@ -19,6 +27,49 @@ public static class PythonHelpers
 
         public static string VirtualEnvironmentPath => Environment.GetEnvironmentVariable("VIRTUAL_ENV");
 
+        /// <summary>
+        /// Determines which Python dependency manager to use based on the files present in the directory.
+        /// Priority: uv (if pyproject.toml + uv.lock) > poetry (if pyproject.toml only) > pip (if requirements.txt)
+        /// </summary>
+        public static PythonDependencyManager DetectPythonDependencyManager(string directory)
+        {
+            var hasPyProjectToml = FileSystemHelpers.FileExists(Path.Combine(directory, Constants.PyProjectToml));
+            var hasUvLock = FileSystemHelpers.FileExists(Path.Combine(directory, Constants.UvLock));
+            var hasRequirementsTxt = FileSystemHelpers.FileExists(Path.Combine(directory, Constants.RequirementsTxt));
```

Comment:
Since we're not checking this till line 53, we should not read the value so early as it might never be used.